Detectors/TRD/base/include/TRDBase/Digit.h

```diff
@@ -13,17 +13,87 @@
 
 #include <cstdint>
 #include <vector>
+#include <array>
+#include <map>
 
 namespace o2
 {
 namespace trd
 {
 
-class DigitIndex;
+class Digit;
 
-typedef std::uint16_t Digit;
-typedef std::vector<Digit> DigitContainer;
-typedef std::vector<DigitIndex> DigitIndexContainer;
+constexpr int kTB = 30;
+constexpr int KEY_MIN = 0;
+constexpr int KEY_MAX = 2211727;
+
+typedef std::uint16_t ADC_t;                              // the ADC value type
+typedef std::array<ADC_t, kTB> ArrayADC_t;                // the array ADC
+typedef std::vector<Digit> DigitContainer_t;              // the digit container type
+typedef std::map<int, ArrayADC_t> ArrayADCMapContainer_t; // a map container type for signal handling during digitization
```

Comment:
could you add an explanation of what the key is this map is supposed to mean? mapping from where to what?

Reply:
The purpose of the map is to fetch signals or create them if they exist or not when a new pad has a signal output, to update or create it. The purpose of the map is that only pads with signals have an element. Therefore, we get rid of the long old `AdcArray` and all the lookup tables to find where a signal should be updated or not.